include/flang/common/real.h

```diff
@@ -0,0 +1,86 @@
+//===-- include/flang/common/real.h -----------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef FORTRAN_COMMON_REAL_H_
+#define FORTRAN_COMMON_REAL_H_
+
+// Characteristics of IEEE-754 & related binary floating-point numbers.
+// The various representations are distinguished by their binary precisions
+// (number of explicit significand bits and any implicit MSB in the fraction).
+
+#include <cinttypes>
+
+namespace Fortran::common {
+
+// Total representation size in bits for each type
+static constexpr int BitsForBinaryPrecision(int binaryPrecision) {
+  switch (binaryPrecision) {
+  case 8: return 16;  // IEEE single (truncated): 1+8+7
+  case 11: return 16;  // IEEE half precision: 1+5+10
+  case 24: return 32;  // IEEE single precision: 1+8+23
+  case 53: return 64;  // IEEE double precision: 1+11+52
+  case 64: return 80;  // x87 extended precision: 1+15+64
+  case 106: return 128;  // "double-double": 2*(1+11+52)
+  case 112: return 128;  // IEEE quad precision: 1+16+111
```

Comment:
Why are numbers prefered for the case values over enums?

Reply:
Do you mean, "why not define an enum to name all of the possible floating-point types?"  All of f18's floating-point constant representations, folding, and decimal conversions are implemented with templates that are parameterized over the binary precisions of the several types.  An enum, if one were present, would just be mapped 1-1 to precision values in all of these use case, so it doesn't seem to be an abstraction that would add value.